Tasks/XamariniOS/xamarinios.ts

```diff
@@ -45,26 +44,18 @@ async function run() {
         const runNugetRestore: boolean = tl.getBoolInput('runNugetRestore');
 
         // find the build tool path based on the build tool and location inputs
-        const buildTool: string = tl.getInput('buildTool');
-        const buildToolLocation: string = tl.getInput('mdtoolLocation', false);
+        const buildToolLocation: string = tl.getInput('buildToolLocation', false);
         let buildToolPath: string;
         if (buildToolLocation) {
             // location is specified
             buildToolPath = buildToolLocation;
-            if (buildTool === 'xbuild' && !buildToolLocation.toLowerCase().endsWith('xbuild')) {
-                buildToolPath = path.join(buildToolLocation, 'xbuild');
-            }
-            if (buildTool === 'msbuild' && !buildToolLocation.toLowerCase().endsWith('msbuild')) {
+            if (buildToolLocation && !buildToolLocation.toLowerCase().endsWith('msbuild')) {
```

Comment:
I can see we want to push the task to use msbuild from the changes to the UI strings and the logic here.  But it appears that this is our logic now:

1. User supplies `path/to/msbuild/directory/msbuild` and we run `path/to/msbuild/directory/msbuild` 
2. User supplies `path/to/msbuild/directory` and we run `path/to/msbuild/directory/msbuild` (task finds executable)
3. User supplies `path/to/xbuild/directory/xbuild` and we run `path/to/xbuild/directory/xbuild`
4. User supplies `path/to/xbuild/directory` and we try to run `path/to/xbuild/directory/msbuild` (and fail)
5. User does not supply `buildToolLocation`.  We find msbuild on the path, else we find xbuild on the path.

Is this correct?  It seems odd to me.  I think it would make more sense to require the user to provide the full path instead of trying to be smart about whether they provided a directory or file.

Reply:
As written, it looks like the task will be confused if the msbuild executable lives in a directory named "msbuild."  So it sounds like you want to `stat` the path to see what it is.

Reply:
In new version of the task we don't support "xbuild" for buildToolLocation at all. I think I will remove this and simplify it to look like this: https://github.com/Microsoft/vsts-tasks/blob/master/Tasks/MSBuild/task.json#L78. In Xamarin iOS first there was mdtool, then xbuild and then msbuild. Hopefully we can stick to msbuild now.